draft-trammell-taps-interface.md

```diff
@@ -1243,28 +1024,93 @@ connectionProperties := Connection.getProperties()
+protocols being used and the system's configuration.
+
+Note that Protocol Properties are also set during pre-establishment, as
+transport parameters, to preconfigure Protocol Stacks during establishment.
+
+Generic Protocol Properties include:
+
+* Timeout for aborting Connection: 
+  This numeric property specifies how long
+  to wait before aborting a Connection during establishment, or after a
+  connection has failed after establishment. It is given in seconds.
+
+* Abort timeout to suggest to the Remote Endpoint: 
+  This numeric property
+  specifies the timeout to propose to the Remote Endpoint. It is given in
+  seconds.
+
+* Retransmission threshold before excessive retransmission notification:
+  This numeric property specifies after how many retransmissions to inform
+  the application about "Excessive Retransmissions".
+
+* Required minimum coverage of the checksum for receiving:
+  This numeric property specifies the part of the received data that needs
+  to be covered by a checksum. It is given in Bytes. A value of 0 means
+  that no checksum is required, and a special value (e.g., -1) indicates
+  full checksum coverage.
+
+* Connection group transmission scheduler: 
+  This enumerated property specifies which
+  scheduler should be used among Connections within a Connection Group. It
+  applies to connection groups; the set of schedulers can be taken from
+  {{I-D.ietf-tsvwg-sctp-ndata}}.
+
+* Maximum message size concurrent with connection establishment:
+  This numeric property represents the maximum Message size that can be sent
+  before or during Connection establishment, see also {{send-idempotent}}.
+  It is given in Bytes. This property is read-only.
+
+* Maximum Message size before fragmentation or segmentation:
+  This numeric property, if applicable, represents the maximum Message size 
+  that can be sent without incurring network-layer fragmentation and/or transport layer segmentation at the sender. This property is read-only.
+
+* Maximum non-partial Message size on send:
+  This numeric property represents the maximum Message size that can be sent as
+  a non-partial Message. This property may be read-only.
+
+* Maximum non-partial Message size on receive: 
+  This numeric property represents
+  the maximum Message size that can be received as a non-partial Message. This property may be read-only.
+
+Specific Protocol Properties can additionally be associated with a specific
+protocol. For example, the application can specify a set of TCP Options to use
+if and only if TCP is selected by the system, including options such as the
+Maximum Segment Size (MSS), and options around Acknowledgement Stretching. Such
+properties should not be assumed to apply across different protocols, but must
+be possible to specify if required by the application. Attempts to set specific
+protocol properties on a protocol stack not containing that specific protocol
+are simply ignored, and do not raise an error.
+
```

Comment:
In issue #69, I proposed the following text as a replacement for this:

***
In order to specify Specific Protocol Properties, Transport System implementations MAY offer applications to attach a set of options to the Preconnection object, associated with a specific protocol. For example, an application could specify a set of TCP Options to use if and only if TCP is selected by the system. Such properties MUST NOT be assumed to apply across different protocols.
***

If you don't agree with the RFC21119 style keywords, don't capitalize them. If you generally don't agree with this text replacement, at least remove ", and options around Acknowledgement Stretching", please.

Reply:
ACK. I'd like to keep the part about not raising errors though...

Reply:
Ah, sure, yes, I missed this in my suggested text update.